python/cudf/cudf/core/column/column.py

```diff
@@ -327,21 +327,23 @@ def to_array(self, fillna=None):
         """
         return self.to_gpu_array(fillna=fillna).copy_to_host()
 
-    def fill(self, fill_value, begin=0, end=-1, inplace=False):
+    def _fill(self, fill_value, begin, end, inplace=False):
+        begin = 0 if begin is None else begin
+        end = self.size if end is None else end
 
-        if end == -1:
-            end = self.size
+        if end < 0:
+            end = self.size + end
 
         assert end <= self.size
         assert begin <= end
```

Comment:
Typical behavior in this case is to just hit the below clause of returning `self.copy()`